utils/cautils/certificate_flow.go

```diff
@@ -286,10 +286,25 @@ func (f *CertificateFlow) CreateSignRequest(ctx *cli.Context, tok, subject strin
 			dnsNames, ips, emails, uris = splitSANs(defaultSANs)
 		}
 	case token.OIDC:
-		if jwt.Payload.Email != "" {
-			emails = append(emails, jwt.Payload.Email)
+		// If no sans are given using the --san flag, and the subject argument
+		// matches the email then CN=token.sub SANs=email, token.iss#token.sub
+		//
+		// If no sans are given and the subject argument does not match the
+		// email then CN=subject SANs=splitSANs(subject)
+		//
+		// If sans are provided CN=subject SANs=splitSANs(sans)
+		if len(sans) == 0 {
+			if jwt.Payload.Email != "" && strings.EqualFold(subject, jwt.Payload.Email) {
+				subject = jwt.Payload.Subject
```

Comment:
In this conditional we're appending SANs to emails and URIs, but in the case below we're creating new splitSANs, ignoring existing SANs. I'm confused by the inconsistency.

Reply:
What are the existing SANs? Due to how tokens work, "sans" is not currently supported on OIDC tokens.

So the only sans are from `--san` or from the `subject` positional argument. 

Without `--san` arguments:
* If the email == subject, then we create the CSR using the same way the CA creates the template, so admins will have the same SANs and CN a non-admin will have.
* If the email != subject, then we create the CSR with the appropriate SAN (dns, email, IP or uri).

But if at least one `--san` is passed, then we create the CSR according to the splitSANs, an admin will get the appropriate certificate, and non-admins will get a certificate with its email in it.

With this logic, admins will always send the CSR that they want, and non-admins too, but they will get a different certificate, the one with the email because that's what the CA does on the OIDC provisioner. The CSR is not checked at all. And this is something that we will have to change in the CA, but as I said in the description this is a different story, and we'll have to change the CA to return an error for non-admins if the CSR has a SAN that does not match the email or uri, or some similar logic.